src/twisted/web/http.py

```diff
@@ -3145,11 +3136,9 @@ def dataReceived(self, data):
         using.
         """
         if self._negotiatedProtocol is None:
-            try:
-                negotiatedProtocol = self._channel.transport.negotiatedProtocol
-            except AttributeError:
-                # Plaintext HTTP, always HTTP/1.1
-                negotiatedProtocol = b"http/1.1"
+            negotiatedProtocol = getattr(
```

Comment:
It's possible there's some edge case where this is an incompatible change?

Reply:
The change looks ok to me.

Was this just a drive-by clenup, or this has a performance impact?

Reply:
About a 2% performance improvement.

Reply:
wow! I was not aware of this. Thanks again for the update